src/app/shared/components/strength-meter/password-strength.ts

```diff
@@ -0,0 +1,70 @@
+/* @format */
+import { Component, Input, OnChanges } from '@angular/core';
+import { FeatureFlagService } from '@root/app/feature-flag/services/feature-flag.service';
+import { passwordStrength } from 'check-password-strength';
+
+enum PasswordStrength {
+  Null = 'null',
+  Weak = 'weak',
+  Medium = 'medium',
+  Strong = 'strong',
+}
+
+@Component({
+  selector: 'pr-password-strength',
+  templateUrl: './password-strength.html',
+  styleUrls: ['./password-strength.scss'],
+})
+export class PasswordStrengthComponent implements OnChanges {
+  progressBars: string[] = ['', '', ''];
+
+  public strength: PasswordStrength = PasswordStrength.Null;
+  public message: string = '';
+  public passwordClass: string = '';
+  @Input() password: string = '';
+  public enabledPasswordCheckStrength: boolean;
+
+  constructor(private feature: FeatureFlagService) {
+    this.enabledPasswordCheckStrength =
+      this.feature.isEnabled('password-strength');
+  }
+
+  ngOnChanges(): void {
+    if (this.enabledPasswordCheckStrength) {
+      this.updatePasswordStrength();
+    }
+  }
+
+  private setStrength(strengthId: number): void {
+    switch (strengthId) {
+      case 0:
+        this.strength = PasswordStrength.Weak;
+        this.message = 'weak';
+        this.passwordClass = 'too-weak';
+        this.progressBars = ['weak', '', ''];
```

Comment:
This is just a general piece of advice. This switch statement is rather verbose, isn't it? This is at least partly due to the fact that while we're now using an enum to declare password strength, we still have to set all of those other values alongside it. We could clear up this code quite a bit if we could derive those values from our `this.strength` value with something like a function call in the typescript layer or a pipe in the template layer.